src/source/Dtls/Dtls.c

```diff
@@ -261,9 +270,22 @@ STATUS createDtlsSession(PDtlsSessionCallbacks pDtlsSessionCallbacks, TIMER_QUEU
         certificateBits = GENERATED_CERTIFICATE_BITS;
     }
 
-    CHK_STATUS(createCertificateAndKey(certificateBits, &(pDtlsSession->pCert), &(pDtlsSession->pKey)));
-    CHK_STATUS(createSslCtx(pDtlsSession->pCert, pDtlsSession->pKey, &(pDtlsSession->pSslCtx)));
-    CHK_STATUS(createSsl(pDtlsSession->pSslCtx, &(pDtlsSession->pSsl)));
+    if (pRtcCertificates == NULL) {
+        CHK_STATUS(createCertificateAndKey(certificateBits, &pDtlsSession->certificates[0].pCert,
+                &pDtlsSession->certificates[0].pKey));
+        pDtlsSession->certificates[0].created = TRUE;
+        pDtlsSession->certificateCount = 1;
```

Comment:
what is the use case for this? when does it happen?

Reply:
The new iteration has changed this but this is the idea of the code generating the cert so we can free ourselves